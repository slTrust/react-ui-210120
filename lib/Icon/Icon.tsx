import React from "react";
import '../importIcons';
import './icon.scss'
interface IconProps {
  name: string;
}

const Icon: React.FC<IconProps> = props => {
  return (
    <span>
      <svg  className="fui-icon">
        <use xlinkHref={`#${props.name}`} />
      </svg>
    </span>
  );
};
export default Icon;
