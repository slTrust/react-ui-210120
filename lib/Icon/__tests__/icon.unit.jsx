import * as renderer from 'react-test-renderer'
import React from 'react'
import Icon from '../Icon.tsx'

describe('Icon', () => {
  it('是个 icon', () => {
    const json = renderer.create(<Icon name="alipay" />)
      .toJSON()
    expect(json).toMatchSnapshot()
  })
})