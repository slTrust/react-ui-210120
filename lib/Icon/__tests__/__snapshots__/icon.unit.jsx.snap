// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon 是个 icon 1`] = `
<svg
  className="fui-icon"
>
  <use
    xlinkHref="#alipay"
  />
</svg>
`;
